### ISSUE TITLE
GitHub Engine

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/scripts#901]
It would be useful to provide an engine implementation for applications hosted on GitHub. Quite a lot of applications can be found on GitHub as source code. Some repositories even provide pre-built executables. A GitHub engine would be responsible for downloading either the source code from the GitHub repository and building the application or for downloading a pre-built executable.

See also #898

**Comment by plata** [comment]
I don't think this is feasible. In general, we cannot ensure that the required dependencies to build or even run an application are available.

**Comment by Kreyren** [comment]
@plata we can assuming we would build it in as container with all required libs

**Comment by plata** [comment]
And who should provide that container?

**Comment by Kreyren** [comment]
i would object for native engine instead of github assuming that the usecase and interpretation is same.
